hcl2template/parser.go

```diff
@@ -210,12 +210,45 @@ func (p *Parser) Parse(filename string, varFiles []string, argVars map[string]st
 	{
 		hclVarFiles, jsonVarFiles, moreDiags := GetHCL2Files(filename, hcl2AutoVarFileExt, hcl2AutoVarJsonFileExt)
 		diags = append(diags, moreDiags...)
+
+		var variableFiles []*hcl.File
+
+		for _, jsonFile := range jsonVarFiles {
+			f, moreDiags := p.ParseJSONFile(jsonFile)
+			diags = append(diags, moreDiags...)
+			if moreDiags.HasErrors() {
+				continue
+			}
+			variableFiles = append(variableFiles, f)
+		}
+
+		for _, hclFile := range hclVarFiles {
+			f, moreDiags := p.ParseHCLFile(hclFile)
+			diags = append(diags, moreDiags...)
+			if moreDiags.HasErrors() {
+				continue
+			}
+			variableFiles = append(variableFiles, f)
+		}
+
+		// We are trying to maintain the user given order for the var files with this for loop.
 		for _, file := range varFiles {
 			switch filepath.Ext(file) {
 			case ".hcl":
-				hclVarFiles = append(hclVarFiles, file)
+				f, moreDiags := p.ParseHCLFile(file)
+				diags = append(diags, moreDiags...)
+				if moreDiags.HasErrors() {
+					continue
+				}
+				variableFiles = append(variableFiles, f)
```

Comment:
This is looking eerily similar between the two different configuration types, I'd think we can re-organise the code to make this error-checking/appending logic common for both types, only calling the right `Parse` function for either type, and forwarding the results to the common parts of the code.